utilities/backupable/backupable_db.cc

```diff
@@ -1201,6 +1211,7 @@ Status BackupEngineImpl::DeleteBackup(BackupID backup_id) {
   }
 
   if (!s1.ok()) {
+    s2.PermitUncheckedError();  // What to do?
```

Comment:
Good question! It looks to me like `GarbageCollect` is optional here? If that is the case, then I think ignoring any `s2` error is fine. If the garbage collection is optional, maybe we should refactor slightly to:

```c++
  auto s1 = DeleteBackupInternal(backup_id);

  // Clean up after any incomplete backup deletion, potentially from
  // earlier session.
  if (might_need_garbage_collect_) {
    GarbageCollect().PermitUncheckedError();  // Garbage Collection is allowed to fail
  }

  if (!s1.ok()) {
    return s1;
  }
  return Status::OK();
```

Reply:
My assumption was a little different.  I assumed that we always needed to do the GarbageCollect, even if the DeleteBackupInternal failed.  If the Delete fails, we return its status code and ignore the code from GarbageCollect.  If the Delete succeeds, we return the code from GarbageCollect.